### DISCUSSION TITLE
Finished with a non-existent file?! Something may have been moved or destroyed during decoding. It is VERY dangerous, avoid it!"

### DISCUSSION THREAD

**Discussion by hayden-t** [savonet/liquidsoap#1889] [Q&A]
https://github.com/savonet/liquidsoap/blob/2b9e35005ee7ed7d50edd21e8f1a26e79d03f787/src/sources/request_source.ml#L88-L101

I delete my file after queueing it via socket connection, it plays fine, but once played I get this error message. Looking at the code it seems there are no other consequences to my actions ? Is it just a false alarm and no harm in deleting after queue ?

**Comment by toots** [comment]
Hi! Files are queued via a request system internally. We are expecting them to be present for the time the request exist in the system. 

If you wish to delete the file after it has been used by liquidsoap, you can use the `tmp` protocol and push a request of this form:
```
tmp:/path/to/file
```
In this case, liquidsoap will take care of deleting the file once it's done using it.

**Reply by hayden-t** [reply]
thank you for your support !! so i can just prefix my queue path with that is simple :) i will try it. but is there any need for the file once its been queued ? and thus any consequence for deleting it ?

**Reply by hayden-t** [reply]
So it did not work:  Unknown protocol "tmp" in URI "tmp:/app/recording.wav"

**Reply by toots** [reply]
This seems to be working here:
```
output.dummy(
  playlist(
    prefix="tmp:",
    "<path/to/playlist>"
    ),
  fallible=true
  )
```

**Reply by toots** [reply]
And, yeah, we need the file to be present to decode data I believe.. 🙂
